### DISCUSSION TITLE
Factory - Pass theme rather than restyling component

### DISCUSSION THREAD

**Discussion by dylan-westbury** [GeekyAnts/NativeBase#4858] [Q&A]
Hi, 

Is it possible to pass the existing styling for a component (e.g. an Input) to a NativeBase factory component, so the styling doesn't have to be recreated and managed separately to the theme? 

For example: 

const NBInput = Factory(TextInput, theme.components.Input);

**Comment by sanna231** [comment]
Hey @dylan-westbury , Yes you can pass. [Here](https://docs.nativebase.io/next/nativebase-factory#page-title) is the example for the same.
